### PR TITLE
[MAINTENANCE] Add back `public_api` decorators to V1-migrated Expectations

### PR DESCRIPTION
**Pull request by cdkini**
Add decorators

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_distinct_values_to_be_in_set.py [hunk 1]
```diff
@@ -48,37 +52,6 @@ class ExpectColumnDistinctValuesToBeInSet(ColumnAggregateExpectation):
...
 
-    For example:
-    ::
-
-        # my_df.my_col = [1,2,2,3,3,3]
-        >>> my_df.expect_column_distinct_values_to_be_in_set(
-                "my_col",
-                [2, 3, 4]
-            )
-        {
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_distinct_values_to_be_in_set.py:59]: Works if we delete this - continuing to debug

File: great_expectations/expectations/core/expect_column_mean_to_be_between.py [hunk 2]
```diff
@@ -90,7 +94,7 @@ class ExpectColumnMeanToBeBetween(ColumnAggregateExpectation):
     strict_max: bool = False
 
     # This dictionary contains metadata for display in the public gallery
-    library_metadata = {
+    library_metadata: ClassVar[dict] = {
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_mean_to_be_between.py:97]: For some reason, these annotations fix build issues

File: great_expectations/expectations/core/expect_column_pair_values_to_be_in_set.py [hunk 3]
```diff
@@ -2,57 +2,19 @@
...
 
     expect_column_pair_values_to_be_in_set is a \
     [Column Pair Map Expectation](https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_pair_map_expectations).
 
-    For example:
-    ::
-        >>> d = {'fruit': ['appple','apple','apple','banana','banana'],
-                'color': ['red','green','yellow','yellow','red']}
-        >>> my_df = pd.DataFrame(data=d)
-        >>> my_df.expect_column_pair_values_to_be_in_set(
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_pair_values_to_be_in_set.py:21]: Works if we delete this - continuing to debug

File: great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py [hunk 4]
```diff
@@ -55,41 +55,17 @@ class QuantileRange(TypedDict):
...
     """Expect the specific provided column quantiles to be between a minimum value and a maximum value.
 
     expect_column_quantile_values_to_be_between is a \
     [Column Aggregate Expectation](https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_aggregate_expectations).
 
-    For example:
-    ::
-
-        # my_df.my_col = [1,2,2,3,3,3,4]
-        >>> my_df.expect_column_quantile_values_to_be_between(
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py:69]: Works if we delete this - continuing to debug

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py [hunk 5]
```diff
@@ -56,17 +56,17 @@ class ExpectColumnValuesToBeInSet(ColumnMapExpectation):
                 "my_col",
                 [2,3]
             )
-        {
+        \{
             "success": false
-            "result": {
+            "result": \{
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_be_in_set.py]: Seems to resolve the matter?

File: great_expectations/expectations/core/expect_column_values_to_not_be_in_set.py [hunk 6]
```diff
@@ -34,28 +34,12 @@
...
 
-class ExpectColumnValuesToNotBeInSet(ColumnMapExpectation):
-    """Expect column entries to not be in the set.
+from great_expectations._docs_decorators import public_api
 
-    For example:
-    ::
 
-        # my_df.my_col = [1,2,2,3,3,3]
-        >>> my_df.expect_column_values_to_not_be_in_set(
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_not_be_in_set.py:44]: Works if we delete this - continuing to debug

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py (hunk 5 above)
Comment (tyler-hoffman) [reply great_expectations/expectations/core/expect_column_values_to_be_in_set.py]: Not saying we definitely shouldn't do this, but:
* This is totally not ergonomic
* Does this work with the gallery build?

Comment (cdkini) [reply great_expectations/expectations/core/expect_column_values_to_be_in_set.py]: I'm not sure to be perfectly honest. I don't like the solution I feel like there's got to be a better way here?

**Review by tyler-hoffman — APPROVED** [review]
LGTM. For anyone wondering - we discussed outside of this PR, and decided that removing the offending code blocks was the best path forward. The curly brackets, which were [the source of the issue](https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3#bad-usage-of-), were barely used, and showed no-longer valid code examples.

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]
Just found a bug!
Screenshot 2024-02-13 at 4 34 04 PM

Looks common to many (all?) of them, but here's the screen I got that from: http://localhost:3000/docs/next/reference/api/expectations/core/ExpectColumnMedianToBeBetween_class

**Pull request closed without merging**